Update Expense Tracker in UserGuide, Glossary and NFR in DeveloperGuide (#102)
AY1920S2-CS2103T-W17-3/main: docs/UserGuide.adoc

* `add n/John Doe p/98765432 e/johnd@example.com a/John street, block 123, #01-01`
* `add n/Betsy Crowe t/friend e/betsycrowe@example.com a/Newgate Prison p/1234567 t/criminal`

=== *Packing List*

Shows a list of all persons in the address book. +
Format: `list`

=== *Expense Manager*

Allow the user to set their budget and track their expenses during their trip.

==== Modifying budgets for a whole trip

The following sections [3.4.1.1 - 3.4.1.3] allows the user to set and modify budgets for a whole trip.

===== Add new budget for a whole trip: `addbudget`

Add a new budget for the whole trip.

> Format: `addbudget amount/AMOUNT [notify/NOTIFYAMOUNT]`

Reviewer: If we were to follow the AB3 format, it should be `NOTIFY_AMOUNT` instead of `NOTIFYAMOUNT`. Same goes for the rest of the format below.
```suggestion
Format: `addbudget amount/AMOUNT [notify/NOTIFY_AMOUNT]`
```

****
* Adds a new budget for a trip.
* Users will be able to set a notification whenever they reach a pre-determined budget. This would be set under `notify/NOTIFYAMOUNT`.

> * The value of `NOTIFYAMOUNT` must be equal to or lesser than `AMOUNT`.

Reviewer: Grammar: "**less** than", not "lesser than"
```suggestion
* The value of `NOTIFY_AMOUNT` must be equal to or less than `AMOUNT`.
```

> * Both `AMOUNT` and `NOTIFYAMOUNT` *must be a positive integer* 1,2,3...

Reviewer: As mentioned above, if we were to follow the AB3 format, it is `NOTIFY_AMOUNT`.

****

Examples:

* `addbudget amount/2000` +
Adds a budget of $2000 to the trip.
* `addbudget amount/2000 notify/1800` +
Adds a budget of $2000 to the trip, and will notify the user when their expenses is at $1800.

===== Edit Budget for a whole trip: `editbudget`

Edit current budget set for the whole trip.

Format: `editbudget amount/AMOUNT [notify/NOTIFYAMOUNT]`

****
* Edits the current budget in place.
* Command can only be used if a budget has been added.
* Existing values will be updated to the input values.
* When editing the budget without specifying an notification amount, the current notification will be removed.
* Any budgets set for specific day will be removed along with their notifications.
* Users will be able to set a notification whenever they reach a pre-determined budget. This would be set under `notify/NOTIFYAMOUNT`.

> * The value of `NOTIFYAMOUNT` must be equal to or lesser than `AMOUNT`.

Reviewer: Grammar: "**less** than", not "lesser than"
```suggestion
* The value of `NOTIFY_AMOUNT` must be equal to or less than `AMOUNT`.
```

* `editbudget amount/5000 notify/4500` +
Edits the current budget of the trip to $5000 to the trip, and will notify the user when their expenses is at $4500.

===== Delete budget for a whole trip: `deletebudget`

Removes budget set for a whole trip.

Format: `deletebudget`

****
* Removes any budget which has been set.
* Command can only be used if a budget has been added.
* Any existing budgets, set for a specific day will be removed as well.
****

Examples:

* `deletebudget` +
Deletes any budgets which has been set for the trip.

==== Setting and modifying budgets for a single day.

The following sections [3.4.2.1 to 3.4.2.3] allows a user to set and modify budgets for a single day of the trip.

===== Add new Budget for a day: `addbudgetday`

Allow the user to add a budget for a specific day of the trip.

Format: `addbudgetday  DAY amount/AMOUNT [notify/NOTIFYAMOUNT]`

****
* Adds a new budget for a single day of the trip at the specified `DAY`.
* The `DAY` selected by the user must not be longer than the length of the trip.
* Both `AMOUNT` and `NOTIFYAMOUNT`specified by the user must be equal to or below the budget set for the whole trip.

> * The value of `NOTIFYAMOUNT` must be equal to or lesser than `AMOUNT`.

Reviewer: Grammar: "**less** than", not "lesser than"
```suggestion
* The value of `NOTIFY_AMOUNT` must be equal to or less than `AMOUNT`.
```

* Users will be able to set a notification whenever they reach a pre-determined budget. This would be set under `notify/NOTIFYAMOUNT`.
* Both `AMOUNT` and `NOTIFYAMOUNT` *must be a positive integer* 1,2,3...
****

Examples:

* `addbudgetday 1 amount/3000` +
Adds a budget for the 1st day of the trip at $3000
* `addbudgetday 1 amount/5000 notify/4500` +
Adds a budget for the 1st day of the trip at $5000 and will notify the user when their expenses on `DAY` 1 is at $4500.

===== Edit new Budget for a day: `editbudgetday`

Allow the user to edit a budget for a specific day of the trip.

Format: `editbudgetday DAY amount/AMOUNT [notify/NOTIFYAMOUNT]`

****
* Edits the current budget for a specific `DAY`. There must be a budget already added for that day.
* The `DAY` selected by the user must not be longer than the length of the trip.
* Existing values will be updated to the input values.
* When editing the budget without specifying an notification amount, the current notification will be removed.

> * The value of `NOTIFYAMOUNT` must be equal to or lesser than `AMOUNT`.

Reviewer: Grammar: "**less** than", not "lesser than"
```suggestion
* The value of `NOTIFY_AMOUNT` must be equal to or less than `AMOUNT`.
```

* `editbudgetday 1 amount/5000 notify/4500` +
Edits the current budget for the 1st day of the trip to $5000, and will notify the user when their expenses is at $4500.

===== Delete budget for a single day: `deletebudgetday`

Allow the user to delete a budget set for a specific day of the trip.

Format: `deletebudgetday DAY`

****
* Removes any budget which has been set for the Day specified by the user.
* Command can only be used if a budget has been set for the day.
****

Examples:

* `deletebudgetday 1` +
Deletes the budget set for the 1st day of the trip.

==== Setting conversion rates: `conversion`

Allow the user to set conversion rates to allow for automatic conversion to SGD for any expenses incurred during the trip.

Format: `conversion rate/RATE`

****
* Sets a conversion rate from their currency to SGD.
* `RATE` must be a positive real number.
* Any expenses entered by the user during the trip would be automatically converted to SGD.
****

==== Modifying expenses

The following sections [3.4.4.1 - 3.4.4.4] allows a user to set and modify expenses incurred during the trip.

===== Adding a new expense: `addexpense`

Allow the user to add a new expense.

> Format: `addexpense DAY detail/DETAIL [amountSGD/AMOUNTSGD] [amoungOTHR/AMOUNTOTHR]`

Reviewer: AB3 format will be: `AMOUNT_SGD` and `AMOUNT_OTHR`

As for prefixes, I think `sgd/` and `othr/` is good enough, but you can use `amountSGD/` as well. 
There is a typo `amoungOTHR/`. I think you meant `amountOTHR/`

```suggestion
Format: `addexpense DAY detail/DETAIL [sgd/AMOUNT_SGD] [other/AMOUNT_OTHER]`
```

****

> * Adds a new expense on specified `DAY`. The day refers to a specific day of the trip. The day *must be a positive integer* and be *lesser than the total number of days of the trip*.

Reviewer: Grammar: "**less** than", not "lesser than"
```suggestion
* Adds a new expense on specified `DAY`. The day refers to a specific day of the trip which *is a positive integer* that is *less than the total number of days of the trip*.
```

****

Examples:

* `addexpense 1 detail/iPHONE11 amountSGD/1100` +
Adds an expense incurred on the 1st day of the trip of an `iPHONE11` for $1100 SGD.

* `addexpense 1 detail/iPHONE11 amountOTHR/800` +
Adds an expense incurred on the 1st day of the trip of an `iPHONE11` for $800. This amount will be converted to SGD using *exchange rate* provided.

===== Editing an expense: `editexpense`

Allow the user to edit an expense. The expense must be already added to the expense tracker.

Format: `editexpense INDEX [detail/DETAIL] [amountSGD/AMOUNTSGD] [amountOTHR/AMOUNTOTHR]`

****
* Edits the expense at the specified `INDEX`. The index refers to the index number shown in the displayed expense list. The index *must be a positive integer* 1, 2, 3, ...
* At least one of the optional fields must be provided.
* Existing values will be updated to the input values.
* An *exchange rate* is required if the user is entering the amount in another currency.
* If the user enters the expense in other currency, the amount will be automatically converted to SGD using the *exchange rate* provided.
****

Examples:

* `editexpense 1 detail/MacbookPro 13 amountSGD/3000` +

> Edit the details and amount paid in SGD of the 1st expense to be `MacbookPro 13` and `$3000 SGD` respectively.

Reviewer: Grammar: "edits", "detail"
```suggestion
Edits the detail and amount paid in SGD of the 1st expense to be `MacbookPro 13` and `$3000 SGD` respectively.
```

* `editexpense 1 detail/AirPods amountOTHR/300` +

> Edit the details and amount paid in other currency of the 1st expense to be `AirPods` and `$300` respectively. The amount will be automatically converted to SGD using *exchange rate* provided.

Reviewer: Grammar: "edits", "detail"
```suggestion
Edits the detail and amount paid in the foreign currency of the 1st expense to be `AirPods` and `$300` respectively. The amount will be automatically converted to SGD using *exchange rate* provided.
```

===== Deleting an expense: `deleteexpense`

> Allow the user to delete an expense. The expense must be already added to the expense tracker.

Reviewer: Grammar: "**Allows** the ..."

Maybe name the feature uniformly? Do you want to call it the `expense manager` or `expense tracker`? Or perhaps `expense tracker` is a sub-feature of `expense manager`?
```suggestion
Allows the user to delete an expense. The expense must be already added to the expense manager.
```

Author: Think I'll change it to expense tracker instead. Since expense tracker is already mentioned in the introduction.

Format: `deleteexpense INDEX`

Examples:
*deleteexpense 1 +
Delete the 1st item in the expense tracker.

===== Listing down expenses for a day: `list`

> Allow the user to see their expense for a specific day of the trip as a list.

Reviewer: Grammar: "**Allows** the ...", "**his/her expenses** for ..."
```suggestion
Shows a list of all expenses for a specified day of the trip.
```

Format: `list DAY`

****

> * Shows a list of expense on a specified `DAY`. The day refers to a specific day of the trip. The day *must be a positive integer* and be *lesser than the total number of days of the trip*.

Reviewer: Grammar: "**less** than", not "lesser than"
```suggestion
* Shows a list of expense on a specified `DAY`. The day refers to a specific day of the trip which *is a positive integer* that is *less than the total number of days of the trip*.
```

* If `DAY` is entered as 0, all expenses incurred during the trip will be shown as a list.
****

Examples:

* `list 1` +
Shows the expenses incurred on the 1st day of the trip.

* `list 0` +
Shows all expenses incurred during the whole duration of the trip

==== Recommend ways to save money based on prevailing usage [Coming in v2.0]

> Use complex algorithms to monitor and track the user's spending habits. Recommend ways to save money on certain items.

Reviewer: Grammar: "**Uses** complex", "**Recommends** ways ..."

Both "user's spending habits" and "user spending habits" should be acceptable.
```suggestion
Uses complex algorithms to monitor and track the user spending habits. Recommends ways to save money on certain items.
```

==== Automatically keep track of expenses [Coming in v2.0]

> Avoid requiring the user to enter their expenses on the application. Instead, all their expenses will be automatically entered into the app for them.

Reviewer: Grammar: "**Avoids** requiring ...", "Instead, all **his/her** expenses ..."
```suggestion
Avoids requiring the user to enter their expenses on the application. Instead, all his/her expenses will be automatically entered into the application for him/her.
```
